### PR TITLE
Clyde EPS service

### PR DESCRIPTION
**Pull request by plauche**
Service wrapper for Clyde EPS API (#62)
- [ ] Test mutations
- [ ] Additional unit/integration tests

### CODE REVIEW

**Comment by plauche** [comment]
[Clydespace EPS Service](https://trello.com/c/d1kt7bqp/221-clydespace-eps-service)

File: services/clyde-3g-eps-service/src/schema/mod.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// limitations under the License.
+//
+
+use kubos_service;
+use models::subsystem::Subsystem;
+
+pub type Context = kubos_service::Context<Subsystem>;
+
+pub mod mutation;
+pub mod query;
```
Comment (CatGarab) [inline services/clyde-3g-eps-service/src/schema/mod.rs]: The queries and mutations available don't match what's in the [service outline doc](http://docs.kubos.co/1.2.0/dev-docs/service-outline-guide.html)

File: services/clyde-3g-eps-service/src/models/subsystem.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        Ok(Subsystem {
+            eps,
+            last_mutation: Cell::new(Mutations::None),
+            errors: RefCell::new(vec![]),
+        })
+    }
+
+    pub fn get_motherboard_telemetry(
+        &self,
+        telem_type: motherboard_telemetry::Type,
```
Comment (CatGarab) [inline services/clyde-3g-eps-service/src/models/subsystem.rs]: Also made a similar comment in your API PR. I dislike that you're specifying the module name in order to clarify which `Type` you're talking about.

File: services/clyde-3g-eps-service/src/models/daughterboard_telemetry.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use clyde_3g_eps_api::DaughterboardTelemetry::Type as DaughterboardTelemetryType;
+
+macro_rules! make_telemetry {
```
Comment (marshall) [inline services/clyde-3g-eps-service/src/models/daughterboard_telemetry.rs]: is there a way to make `clyde_3g_eps_api::DaughterboardTelemetry::Type` Serializable/Derserializable into a string so we can avoid manually maintaining this mapping just for juniper?

same question below for `MotherboardTelemetry::Type`, and `ResetTelemetry::Type`

File: services/clyde-3g-eps-service/src/models/subsystem.rs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            .get_motherboard_telemetry(telem_type.into())?)
+    }
+
+    pub fn get_daughterboard_telemetry(
+        &self,
+        telem_type: daughterboard_telemetry::Type,
+    ) -> Result<f32, Error> {
+        Ok(self.eps
+            .lock()
+            .unwrap()
```
Comment (marshall) [inline services/clyde-3g-eps-service/src/models/subsystem.rs]: should these calls to `unwrap()` also be guarded by `?`

**Comment by plauche** [comment]
The query side of this PR is ready for re-review @CatGarab.

File: apis/clyde-3g-eps-api/src/commands/watchdog.rs [hunk 5]
```diff
@@ -48,7 +48,7 @@ pub mod get_comms_watchdog_period {
     use super::*;
 
     pub fn parse(data: &[u8]) -> EpsResult<u8> {
-        if data.len() == 2 {
+        if data.len() == 2 || data.len() == 4 {
```
Comment (CatGarab) [inline apis/clyde-3g-eps-api/src/commands/watchdog.rs]: The manual says this only ever returns 2 bytes

File: services/clyde-3g-eps-service/src/bin/clyde-3g-eps-service.rs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+//! ```toml
+//! [clyde-3g-eps-service]
+//! bus = "i2c-1"
+//!
+//! [clyde-3g-eps-service.addr]
+//! ip = "127.0.0.1"
+//! port = 8089
+//! ```
+//!
+//! Where `bus` specifies the I2c bus the eps is on, `ip` specifies the
```
Comment (CatGarab) [inline services/clyde-3g-eps-service/src/bin/clyde-3g-eps-service.rs]: nit: capitalize I2C and EPS :stuck_out_tongue:

File: services/clyde-3g-eps-service/src/bin/clyde-3g-eps-service.rs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+//! Service for interacting with a ClydeSpace 3G EPS
+//!
+//! # Configuration
+//!
+//! The service can be configured in the `/home/system/etc/config.toml` with the following fields:
+//!
+//! ```toml
+//! [clyde-3g-eps-service]
+//! bus = "i2c-1"
```
Comment (CatGarab) [inline services/clyde-3g-eps-service/src/bin/clyde-3g-eps-service.rs]: nit: This should be "/dev/i2c-1"

File: services/clyde-3g-eps-service/src/schema/query.rs [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    //                 revision: i32,
+    //                 firmwareVersion: i32
+    //             },
+    //             daughterboard {
+    //                 revision: i32,
+    //                 firmwareVersion: i32
+    //             }
+    //         },
+    //         watchdogPeriod: i32,
+
```
Comment (CatGarab) [inline services/clyde-3g-eps-service/src/schema/query.rs]: nit: extra line

File: services/clyde-3g-eps-service/src/schema/query.rs [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    //     }
+    // }
```
Comment (CatGarab) [inline services/clyde-3g-eps-service/src/schema/query.rs]: This doesn't include all of the telemetry fields (missing reset, lastEpsError, and boardStatus).

**Pull request merged**